task_reactredux/task1/src/reducers/MailReducer.js

```diff
@@ -14,9 +14,14 @@ const mailReducer = (state = initialState, action) => {
                 mail: action.mail,
                 mailValid: validateMail(action.mail)
             }
+        case SAVE_MAIL_FROM_FORM:
+            return {
+                ...state,
+                mail: action.mail
```

Comment:
If for some reason you will need `mailValid` field on other pages, it's a good idea to set this field to `true`. Same applies to password